### DISCUSSION TITLE
vlt package manager support

### DISCUSSION THREAD

**Discussion by rarkins** [renovatebot/renovate#32633] [Suggest an Idea]
### Tell us more.

Introduced here: https://blog.vlt.sh/blog/introducing-vlt-and-vsr

It has backwards-compatibility with npm's `package.json`, e.g. when I run `vlt install` in a simple npm project then it adds a `vlt-lock.json` file. In that sense initial support could be similar to pnpm/yarn.

Important for us to understand if the structure of the `vlt-lock.json` file is stable enough that we won't be "broken" by changes to it too often. Alternatively if parsing of the lock file is exposed by `vlt` as a JS library then we could use that instead.

There's probably also not much point in us only supporting `vlt` users who use it as a drop-in npm replacement. It's important to understand and support `vlt`'s method of configuring registries, aliases, vsr, etc.

Cc @darcyclarke

**Comment by ruyadorno** [comment]
> Important for us to understand if the structure of the `vlt-lock.json` file is stable enough that we won't be "broken" by changes to it too often. 

At the moment the `vlt` client is still in `v0.0.0` and very much experimental, so things can definitely still radically change.

> Alternatively if parsing of the lock file is exposed by `vlt` as a JS library then we could use that instead.

Yes, we do have a programmatic API available, so that it would be possible to use it to parse, e.g:

```
import { lockfile } from '@vltpkg/graph'

lockfile.load({
  projectRoot: 'path/to/project',
  manifest: parsedPackageJsonObject,
})
```

But again, everything is still experimental and I would advise on waiting for v1 in case you want to avoid the extra churn of maintaining an API that is in active development.

Regardless, we'll be very happy to support you here :)
